src/ipmanager/api/views.py

```diff
@@ -47,45 +47,47 @@ def get(self, request, group_key):
             return JsonResponse(body, status=404)
         
 
-class APICheckView(View):
+class CheckView(View):
 
   def get(self, request):
 
     # Made-up data
     group_keys = ["A", "B", "C", "D"]
 
-    # Using query_string (unparsed) and to create Query Dict object to extrack parameters
-    dict = QueryDict(query_string = request.META['QUERY_STRING'], mutable=False, encoding=None)
-    
     # Returning 400 error if wrong parameters or ip not present
-    if 'ip' not in dict:
-      return JsonResponse({"title": "Unparseable IP address", 
-                           "detail": "The value 'invalid_ip' you provided in the 'ip' query parameter cannot be" + 
-                                     "parsed as a valid IP address.", 
+    if 'ip' not in request.GET:
+      return JsonResponse({"title": "IP address not provided", 
+                           "detail": "IP address was not provided as a query parameter", 
                            "status": 400}, status=400)
     
-    # Can extract ip from dict now
-    ip = dict['ip']
+    ip = request.GET['ip']
+    try :
+        IPv4Address(ip)
+    except AddressValueError:
+        return JsonResponse({"title": "Unparseable IP address", 
+                             "detail": "The value 'invalid_ip' you provided in the 'ip' query parameter cannot be " + 
```

Comment:
You should insert the actual submitted value in place of "invalid_ip". A good way to do this is using Python [formatted string literals](https://docs.python.org/3/tutorial/inputoutput.html#tut-f-strings) (a.k.a., "f-strings"):

```python
f"The value '{ip}' you provided in the 'ip' query parameter cannot be "
```